src/ui/zcl_abapgit_gui_page_patch.clas.abap

```diff
@@ -1,798 +1,797 @@
+      IMPORTING
+        !it_diff_files_old TYPE tt_file_diff
+      RAISING
+        zcx_abapgit_exception .
+    METHODS add_to_stage
+      RAISING
+        zcx_abapgit_exception .
+    METHODS refresh
+      IMPORTING
+        !iv_action TYPE clike
+      RAISING
+        zcx_abapgit_exception .
+    METHODS refresh_full
+      RAISING
+        zcx_abapgit_exception .
+    METHODS refresh_local
+      RAISING
+        zcx_abapgit_exception .
+    METHODS refresh_local_object
+      IMPORTING
+        !iv_action TYPE clike
+      RAISING
+        zcx_abapgit_exception .
+    METHODS apply_patch_all
+      IMPORTING
+        !iv_patch      TYPE string
+        !iv_patch_flag TYPE abap_bool
+      RAISING
+        zcx_abapgit_exception .
+    METHODS are_all_lines_patched
+      IMPORTING
+        !it_diff                        TYPE zif_abapgit_definitions=>ty_diffs_tt
+      RETURNING
+        VALUE(rv_are_all_lines_patched) TYPE abap_bool .
+    METHODS apply_patch_for
+      IMPORTING
+        !iv_filename   TYPE string
+        !iv_line_index TYPE string
+        !iv_patch_flag TYPE abap_bool
+      RAISING
+        zcx_abapgit_exception .
+    METHODS get_diff_object
+      IMPORTING
+        !iv_filename   TYPE string
+      RETURNING
+        VALUE(ro_diff) TYPE REF TO zcl_abapgit_diff
+      RAISING
+        zcx_abapgit_exception .
+    METHODS get_diff_line
+      IMPORTING
+        !io_diff       TYPE REF TO zcl_abapgit_diff
+        !iv_line_index TYPE string
+      RETURNING
+        VALUE(rs_diff) TYPE zif_abapgit_definitions=>ty_diff
+      RAISING
+        zcx_abapgit_exception .
+    METHODS is_every_changed_line_patched
+      RETURNING
+        VALUE(rv_everything_patched) TYPE abap_bool .
+    CLASS-METHODS is_patch_line_possible
+      IMPORTING
+        !is_diff_line                    TYPE zif_abapgit_definitions=>ty_diff
+      RETURNING
+        VALUE(rv_is_patch_line_possible) TYPE abap_bool .
+    METHODS render_scripts
+      RETURNING
+        VALUE(ro_html) TYPE REF TO zcl_abapgit_html
+      RAISING
+        zcx_abapgit_exception .
+ENDCLASS.
+
+
+
+CLASS ZCL_ABAPGIT_GUI_PAGE_PATCH IMPLEMENTATION.
+
+
+  METHOD add_menu_begin.
+
+    io_menu->add(
+        iv_txt   = |Refresh Local|
+        iv_typ   = zif_abapgit_html=>c_action_type-dummy
+        iv_act   = c_actions-refresh_local
+        iv_id    = c_actions-refresh_local
+        iv_title = |Refresh all local objects, without refreshing the remote| ).
+
+    io_menu->add(
+        iv_txt   = |Refresh|
+        iv_typ   = zif_abapgit_html=>c_action_type-dummy
+        iv_act   = c_actions-refresh
+        iv_id    = c_actions-refresh
+        iv_title = |Complete refresh of all objects, local and remote| ).
+
+  ENDMETHOD.
+
+
+  METHOD add_menu_end.
+
+    io_menu->add( iv_txt = 'Stage'
+                  iv_act = c_actions-stage
+                  iv_id  = 'stage'
+                  iv_typ = zif_abapgit_html=>c_action_type-dummy
 ).
```

Comment:
there's some strange character here

Reply:
yea, I'll have a look tomorrow, probably something wrong with the quickfix, plus guess the linter should also give a parser error

Reply:
stupid carriage returns, logic for catching this early implemented in https://github.com/abaplint/abaplint/pull/1269/commits/52a7661a44b6ab92ab3229e9d6b1f3f7fd07f6cf